src/test/daml/Daml/Finance/Instrument/Equity/Test/Dividend.daml

```diff
@@ -147,4 +147,74 @@ run = script do
   Holding.verifyOwnerAndAmountOfHolding [(investor, 1000.25, investorEquityHoldingCid),
     (investor, 2000.50, investorCashHoldingCid)]
 
+  -- Test bonus issue (works in a similar way to the dividend above)
+
+  -- Distribute holdings: fractional holdings are also supported.
+  investorEquityCid <- Account.credit [publicParty] cumEquityInstrument 1000.25 investorAccount
+
+  -- CREATE_EQUITY_BONUS_ISSUE_DISTRIBUTION_EVENT_BEGIN
+  -- Create bonus issue event: receive 2 additional shares for each share currently held
+  -- (this also works with fractional shares)
+  distributionEventCid <-
+    Instrument.submitExerciseInterfaceByKeyCmd @Equity.I [issuer] [] cumEquityInstrument
+      Equity.DeclareDividend with
+        id = Id $ "APPL - " <> show now
+        description = "Bonus issue"
+        effectiveTime = now
+        newInstrument = exEquityInstrument
+        perUnitDistribution = [qty 2.0 exEquityInstrument]
+  -- CREATE_EQUITY_BONUS_ISSUE_DISTRIBUTION_EVENT_END
+
+    -- Lifecycle bonus issue
+  (_, [effectCid]) <- submit issuer do
+    exerciseCmd distributionRuleCid Lifecycle.Evolve with
+      observableCids = []
+      eventCid = distributionEventCid
+      instrument = cumEquityInstrument
+
+  -- Claim effect (enable netting so that there is only one holding with the new correct quantity)
+  claimRuleCid <- toInterfaceContractId @Claim.I <$> submit custodian do
```

Comment:
do we need a new rule?

Reply:
I wanted to do netting in the bonus issue case (to get the total stocks as one holding), but not necessarily for the cash dividend.

Reply:
understood, thx! 

If you net on the cash dividend that should not be an issue, as it's different instruments

Reply:
👍